www/js/control/ProfileSettings.jsx

```diff
@@ -332,14 +594,110 @@ const ProfileSettings = () => {
+                        </Button>
+                        <Button onPress={()=>{
+                            setDataPendingVis(false);
+                            forceSync();}}>
+                                {t('general-settings.confirm')}
+                        </Button>
+                    </Dialog.Actions>
+                </Dialog>
+            </Modal>
+
+            {/* handle no consent */}
+            <Modal visible={noConsentVis} onDismiss={()=>setNoConsentVis(false)} transparent={true}>
+                <Dialog visible={noConsentVis} 
+                        onDismiss={()=>setNoConsentVis(false)} 
+                        style={styles.dialog(colors.elevation.level3)}>
+                    <Dialog.Title>{t('general-settings.consent-not-found')}</Dialog.Title>
+                    <Dialog.Actions>
+                        <Button onPress={()=>{
+                            setNoConsentVis(false);
+                            setNoConsentMessageVis(true)}}>
+                                {t('general-settings.cancel')}
+                        </Button>
+                        <Button onPress={()=>{
+                            setNoConsentVis(false);
+                            // $state.go("root.reconsent"); //don't know how to do this yet
+                            }}>
+                                {t('general-settings.confirm')}
+                        </Button>
+                    </Dialog.Actions>
+                </Dialog>
+            </Modal>
+
+            {/* handle consent */}
+            <Modal visible={consentVis} onDismiss={()=>setConsentVis(false)} transparent={true}>
+                <Dialog visible={consentVis} 
+                        onDismiss={()=>setConsentVis(false)} 
+                        style={styles.dialog(colors.elevation.level3)}>
+                    <Dialog.Title>{t('general-settings.consented-to', {protocol_id: consentDoc.protocol_id, approval_date: consentDoc.approval_date})}</Dialog.Title>
+                    <Dialog.Actions>
+                        <Button onPress={()=>{
+                            setConsentDoc({});
+                            setConsentVis(false);}}>
+                                {t('general-settings.consented-ok')}
+                        </Button>
+                    </Dialog.Actions>
+                </Dialog>
+            </Modal>
+
+            <DataDatePicker date={dumpDate} setDate={setDumpDate} open={dateDumpVis} setOpen={setDateDumpVis}></DataDatePicker>
+
+            <AlertBar visible={dataPushedVis} setVisible={setDataPushedVis} messageKey='all data pushed!'></AlertBar>
+            <AlertBar visible={invalidateSuccessVis} setVisible={setInvalidateSuccessVis} messageKey='success -> ' messageAddition={cacheResult}></AlertBar>
+            <AlertBar visible={noConsentMessageVis} setVisible={setNoConsentMessageVis} messageKey='general-settings.no-consent-message'></AlertBar> 
+        
```

Comment:
all of the `Modal` and `AlertBar` components replace ionic popups and their visibility is controlled by a state variable